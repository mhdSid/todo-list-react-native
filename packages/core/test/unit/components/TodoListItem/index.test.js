import { render } from '@testing-library/react-native'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import Component from '../../../../src/components/TodoListItem'
import selectors from '../../../lib/selector/todoListItem'
import MockThemeProvider from '../../../lib/mocks/MockThemeProvider'

const mockStore = configureStore([thunk])

const store = mockStore({
  todoList: {
    list: []
  },
  auth: {
    user: {
      isLoggedIn: false
    }
  },
  theme: {
    theme: 'light'
  }
})

describe('TodoListItem', () => {
  test('root exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const root = await element.findByTestId(selectors.root)
    expect(root).toBeDefined()
  })
  test('checkbox exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const checkbox = await element.findByTestId(selectors.checkbox)
    expect(checkbox).toBeDefined()
  })
  test('task exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const task = await element.findByTestId(selectors.task)
    expect(task).toBeDefined()
  })
  test('snapshot', () => {
    const data = render(MockThemeProvider(Component, store)).toJSON()
    expect(data).toMatchSnapshot()
  })
})
