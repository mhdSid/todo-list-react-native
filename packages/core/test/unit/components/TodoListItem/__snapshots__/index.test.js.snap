// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem snapshot 1`] = `
<View
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "padding": 20,
      },
      Object {
        "backgroundColor": "rgba(1,1,1,0.1)",
      },
    ]
  }
  testID="todoListItem.root"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": false,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "#657786",
            "borderRadius": 2,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          Object {
            "marginRight": 20,
          },
          false,
          false,
          undefined,
          false,
        ]
      }
      testID="todoListItem.checkbox"
    />
    <Text
      style={
        Object {
          "color": "#333",
        }
      }
      testID="todoListItem.task"
    />
  </View>
</View>
`;
